OSDOCS-12134: 4.14.38 z-stream RN (#82995)
openshift/openshift-docs: release_notes/ocp-4-14-release-notes.adoc

====
For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====
// 4.14.38
=== RHSA-2024:7184 - {product-title} 4.14.38 bug fix and security update

Issued: 3 October 2024

{product-title} release 4.14.38, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:7184[RHSA-2024:7184] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:7187[RHSA-2024:7187] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.14.38 --pullspecs
----

[id="ocp-4-14-38-enhancements"]
==== Enhancements

The following enhancement is included in this z-stream release:

[id="ocp-4-14-38-enhancements-_{context}"]
===== Backporting configurable subnets

> * This release includes configurable subnets that allow you to configure the masquerade, as well as join and transition switch subnets to prevent overlaps with ip addresses that are used in your local infrastructure. (link:https://issues.redhat.com/browse/OCPBUGS-38440[*OCPBUGS-38440*])

Reviewer: ```suggestion
* This release includes configurable subnets that you can use to configure the masquerade. You can also use configurable subnets to join and transition subnets to prevent overlaps with IP addresses that are used in your local infrastructure. (link:https://issues.redhat.com/browse/OCPBUGS-38440[*OCPBUGS-38440*])
```

Reviewer: "Allow" is a form of anthrophomorphism.

![Screenshot from 2024-10-03 17-15-40](https://github.com/user-attachments/assets/0cff9545-d39c-4689-b4bb-2121824f15d9)

[id="ocp-4-14-38-bug-fixes_{context}"]
==== Bug fixes

> * Previously, an `AdditionalTrustedCA` field specified in the Hosted Cluster image configuration was not reconciled into the openshift-config namespace as expected and the component was unavailable. With this release, the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-42184[*OCPBUGS-42184*])

Reviewer: ```suggestion
* Previously, an `AdditionalTrustedCA` field specified in the Hosted Cluster image configuration was not reconciled into the `openshift-config` namespace as expected and the component was unavailable. With this release, the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-42184[*OCPBUGS-42184*])
```

> * Previously, Operator Lifecycle Manager (OLM) catalog source pods did not recover from node failure if the `registryPoll` field was `none`. With this release, OLM `CatalogSource` registry pods recover from cluster node failures and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-42150[*OCPBUGS-42150*])

Reviewer: ```suggestion
* Previously, {olm-first} catalog source pods did not recover from node failure if the `registryPoll` field was `none`. With this release, OLM `CatalogSource` registry pods recover from cluster node failures and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-42150[*OCPBUGS-42150*])
```

Reviewer: ![Uploading Screenshot from 2024-10-03 17-17-25.png…]()

* Previously, if you created a hosted cluster by using a proxy to make the cluster reach a control plane from a compute node, the compute node would be unavailable to the cluster. With this release, the proxy settings are updated for the node so that the node can use a proxy to communicate with the control plane successfully. (link:https://issues.redhat.com/browse/OCPBUGS-42021[*OCPBUGS-42021*])

> * Previously, when the Operator Lifecycle Manager (OLM) evaluated a potential upgrade, it used the dynamic client list for all custom resource (CR) instances in the cluster. Clusters with a large number of CRs could experience timeouts from the apiserver and stranded upgrades. With this release, the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-42017[*OCPBUGS-42017*])

Reviewer: ```suggestion
* Previously, when the {olm-first} evaluated a potential upgrade, it used the dynamic client list for all custom resource (CR) instances in the cluster. Clusters with a large number of CRs could experience timeouts from the `apiserver` and stranded upgrades. With this release, the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-42017[*OCPBUGS-42017*])
```

* Previously, the {product-title} web console *Topology* view could only display a maximum of 100 nodes. If you attempted to view more than 100 nodes, the web console would output a `Loading is taking longer than expected.` error message. With this release, the `MAX_NODES_LIMIT` parameter for the web console is set to `200` so that the web console can display a maximum of 200 nodes. (link:https://issues.redhat.com/browse/OCPBUGS-41581[*OCPBUGS-41581*]

* Previously, when you configured a hosted cluster to use an identity provider (IdP) that has either an `http` or `https` endpoint, the IdP hostname did not resolve when sent through the proxy. With this release, a DNS lookup operation checks the IdP before IdP traffic is sent through a proxy, so that IdPs with hostnames can only be resolved by the data plane and verified by the Control Plane Operator (CPO). (link:https://issues.redhat.com/browse/OCPBUGS-41374[*OCPBUGS-41374*])

> 

Reviewer: Remove the extra space?

> * Previously, the Cloud Credential Operator (CCO) would produce an error when starting or restarting when a large number of secrets in the cluster were fetched at once. With this release, CCO fetches the secrets in batches of 100 and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-41236[*OCPBUGS-41236*])

Reviewer: ```suggestion
* Previously, the Cloud Credential Operator (CCO) would produce an error when starting or restarting a large number of secrets in the cluster. The CCO attempted to fetch the secrets at the same time. With this release, CCO fetches the secrets in batches of 100 and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-41236[*OCPBUGS-41236*])
```

> //* Previously, the grace period for a node to become ready was not aligned with upstream behavior. Occasionally, the grace period caused a node to cycle between `Ready` and `Not ready` states. With this release, the issue is fixed so that the grace period does not cause a node to cycle between the two states. (link:https://issues.redhat.com/browse/OCPBUGS-39378[*OCPBUGS-39378*])

Reviewer: This note is pending an SME ack or can it be removed?

Author: It's pending the removal of the red hat security label :)

* Previously, the `openvswitch` service used older cluster configurations after a cluster upgrade and this caused the `openvswitch` service to stop. With this release, the `openvswitch` service is now restarted after a cluster upgrade so that the service uses the newer cluster configurations. (link:https://issues.redhat.com/browse/OCPBUGS-39192[*OCPBUGS-39192*])

* Previously, proxying for Operators that run in the control plane of a hosted cluster was performed through proxy settings on the Konnectivity agent pod that runs in the data plane. As a consequence, it was not possible to distinguish whether proxying was needed based on application protocol. 
+

> For parity with OpenShift Container Platform, IDP communication via HTTPS or HTTP should be proxied, but LDAP communication should not be proxied. This type of proxying also ignores `NO_PROXY` entries that rely on host names because by the time traffic reaches the Konnectivity agent, only the destination IP address is available. With this release, in hosted clusters, proxy is invoked in the control plane via `konnectivity-https-proxy` and `konnectivity-socks5-proxy`, and proxying traffic is stopped from the Konnectivity agent. As a result, traffic that is destined for LDAP servers is no longer proxied. Other HTTPS or HTTPS traffic is proxied correctly. The `NO_PROXY` setting is honored when you specify hostnames. (link:https://issues.redhat.com/browse/OCPBUGS-38066[*OCPBUGS-38066*])

Reviewer: ```suggestion
For parity with {product-title}, IdP communication through HTTPS or HTTP protocols must be proxied, but LDAP communication must not be proxied. This type of proxying also ignores `NO_PROXY` entries that rely on host names because by the time traffic reaches the Konnectivity agent. This means that  only the destination IP address is available. With this release, in hosted clusters, proxy is invoked in the control plane through `konnectivity-https-proxy` and `konnectivity-socks5-proxy`, so that proxying traffic is stopped from the Konnectivity agent. As a result, traffic that is destined for LDAP servers is no longer proxied. Other HTTPS or HTTPS traffic is proxied correctly. The `NO_PROXY` setting is honored when you specify hostnames. (link:https://issues.redhat.com/browse/OCPBUGS-38066[*OCPBUGS-38066*])
```

* Previously, proxying for IDP communication occurred in the Konnectivity agent. By the time traffic reached Konnectivity, its protocol and hostname were no longer available. Consequently, proxying was not done correctly for the OAUTH server pod. It did not distinguish between protocols that require proxying (HTTP or HTTPS) and protocols that do not (LDAP). In addition, it did not honor the `no_proxy` variable that is configured in the `HostedCluster.spec.configuration.proxy` spec. 
+
With this release, you can configure the proxy on the Konnectivity sidecar of the OAUTH server so that traffic is routed appropriately, honoring your `no_proxy` settings. As a result, the OAUTH server can communicate properly with identity providers when a proxy is configured for the hosted cluster. (link:https://issues.redhat.com/browse/OCPBUGS-38060[*OCPBUGS-38060*]

[id="ocp-4-14-38-known-issue_{context}"]
==== Known issues

> * Deploying a self-managed private hosted cluster on AWS fails because the `bootstrap-kubeconfig` file uses an incorrect KAS port. As a result, the AWS instances are provisioned but cannot join the hosted cluster as nodes. (link:https://issues.redhat.com/browse/OCPBUGS-42221[*OCPBUGS-42221*])

Reviewer: I think KAS means the `api-server`.
